src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -495,6 +442,71 @@ protected override void Dispose(bool disposing)
             return context;
         }
 
+        private static ScriptFunc CreateLoggerFunc(WeakReference<TraceWriter> traceWriterReference, WeakReference<FunctionInvocationContext> invocationContextReference)
+        {
+            return (ScriptFunc)(p =>
+            {
+                var logData = (IDictionary<string, object>)p;
+                string message = (string)logData["msg"];
+                if (message != null)
+                {
+                    try
+                    {
+                        // Node captures the AsyncLocal value of the first invocation, which means that logs
+                        // are correlated incorrectly. Here we'll overwrite that value with the correct value
+                        // immediately before logging.
+                        TraceWriter resolvedTraceWriter = null;
+                        FunctionInvocationContext resolvedInvocationContext = null;
+                        if (traceWriterReference.TryGetTarget(out resolvedTraceWriter) &&
+                            invocationContextReference.TryGetTarget(out resolvedInvocationContext))
+                        {
+                            using (BeginFunctionScope(resolvedInvocationContext))
+                            {
+                                // TraceWriter already logs to ILogger
+                                TraceLevel level = (TraceLevel)logData["lvl"];
+                                var evt = new TraceEvent(level, message);
+                                resolvedTraceWriter.Trace(evt);
+                            }
+                        }
+                    }
+                    catch (ObjectDisposedException)
+                    {
+                        // if a function attempts to write to a disposed
+                        // TraceWriter. Might happen if a function tries to
+                        // log after calling done()
+                    }
+                }
+
+                return Task.FromResult<object>(null);
```

Comment:
nit: Use `Task.CompletedTask`

Reply:
That doesn't work because this is a Task<object> not Task

Reply:
👍 
Just looked at this line (not the signature) and assumed this was just using an old pattern from the existing code.